crates/bevy_math/src/lerp.rs

```diff
@@ -0,0 +1,56 @@
+/// Provides a trait to implement the linear interpolation function.
+pub trait Lerp {
+    /// The Scalar type used in the linear interpolation function.
+    type Scalar;
+
+    /// Performs a linear interpolation between `self` and `rhs` based on the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s` is `1.0`, the result
+    /// will be equal to `rhs`. When `s` is outside of range `[0, 1]`, the result is linearly
+    /// extrapolated.
+    fn lerp(self, rhs: Self, s: Self::Scalar) -> Self;
+}
+
+// implementation for floats
+impl Lerp for f32 {
+    type Scalar = f32;
+
+    #[inline(always)]
+    fn lerp(self, rhs: Self, s: Self::Scalar) -> Self {
+        self + ((rhs - self) * s)
+    }
+}
+
+impl Lerp for f64 {
+    type Scalar = f64;
+
+    #[inline(always)]
+    fn lerp(self, rhs: Self, s: Self::Scalar) -> Self {
+        self + ((rhs - self) * s)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
What about lerping for the integer types?

Reply:
I didn't implement them. I assumed the user could implement them if needed.

Reply:
Due to Rust's coherence rules the user can't implement `Lerp` on integer types directly. It's better if bevy implemented `Lerp` on stdlib types as much as possible where it make sense.